src/Tools/ExternalAccess/OmniSharp/NavigateTo/OmniSharpNavigateToSearchService.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Navigation;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.NavigateTo;
+
+internal static class OmniSharpNavigateToSearcher
+{
+    public delegate Task OmniSharpNavigateToCallback(Project project, in OmniSharpNavigateToSearchResult result, CancellationToken cancellationToken);
+
+    public static Task SearchAsync(
+        Solution solution,
+        OmniSharpNavigateToCallback callback,
+        string searchPattern,
+        IImmutableSet<string> kinds,
+        CancellationToken cancellationToken)
+    {
+        var searcher = NavigateToSearcher.Create(
+            solution,
+            AsynchronousOperationListenerProvider.NullListener,
```

Comment:
This is the part I'm not sure what to do with. @CyrusNajmabadi, is this the right thing to do?

Reply:
this is fine.  you're unlikely to be testing NavTo with integration tests.  if you did, we'd need to thread this through properly.